Extensions/GoogleAnalyticsExperiments/Src/UIContribution/src/ExperimentRESTClient.tsx

```diff
@@ -0,0 +1,69 @@
+import TFS_ServiceEndpoint_Contracts = require("TFS/ServiceEndpoint/Contracts");
+import TFS_ServiceEndpoint_TaskAgentClient = require("TFS/ServiceEndpoint/ServiceEndpointRestClient");
+
+export class ExperimentRESTClient {
+	public async getExperiment<T>(endpointId: string, experimentId: string, accountId: string, webPropertyId: string, profileId: string): Promise<T> {
+		const parameters = {
+			accountId ,
+			webPropertyId,
+			profileId,
+			experimentId };
+		const dataSourceName = "fetchExperiment";
+		const experimentResults = await this.getDatasourceResponse(endpointId, dataSourceName, parameters);
+		return experimentResults;
+	}
+
+	public async getExperimentResult< T >( endpointId: string, experimentId: string, ids: string, startDate: string, endDate: string, metrics: string, dimensions: string ): Promise<T> {
+		const parameters = {
+			ids ,
+			startDate ,
+			endDate,
+			metrics,
+			dimensions};
+		const dataSourceName = "fetchData";
+		const results = await this.getDatasourceResponse(endpointId, dataSourceName, parameters) ;
+
+		const experimentResults = results.filter((row) =>  row[0] === experimentId );
```

Comment:
can row be null or empty array?

Reply:
We are already checking that in "getDatasourceResponse" method.